cms/static/js/views/metadata.js

```diff
@@ -220,6 +225,44 @@ function(BaseView, _, MetadataModel, AbstractEditor, FileUpload, UploadDialog, V
 
     });
 
+    Metadata.NumberWithWarning = Metadata.Number.extend({
+        initialize: function() {
+            Metadata.Number.prototype.initialize.apply(this);
+            var options = this.model.get("options");
+            this.msg = options["warning"];
+            // show initial warning if necessary
+            this.checkWarning();
+        },
+
+        changed: function() {
+            Metadata.Number.prototype.changed.apply(this);
+            this.checkWarning();
+        },
+
+        checkWarning: function() {
+            var value = this.getValueFromEditor();
+            if (parseInt(value)) {
```

Comment:
One tricky part of `parseInt` is that it will ignore leading whitespace, read in digits and then ignore trailing non-digits.

Consider:
```javascript
parseInt('       100 bottles of beer')
// === 100
// Truthy

parseInt('0mg!')
// 0
// Falsey
```

Could this `if` clause be affected by this behavior?